data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/SearchAccessorStrategy.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source.opensearch.worker.client;
+
+import org.opensearch.dataprepper.plugins.source.opensearch.OpenSearchSourceConfiguration;
+
+/**
+ * SearchAccessorStrategy determines which {@link SearchAccessor} (Elasticsearch or OpenSearch) should be used based on
+ * the {@link OpenSearchSourceConfiguration}.
+ * @since 2.4
+ */
+public class SearchAccessorStrategy {
```

Comment:
Is this class replacing the `SearchClientFactory` from the class diagram?

Reply:
Yes, I forgot I renamed it. Good catch.